configure_cmd.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"fmt"
+	"github.com/briandowns/spinner"
+	"github.com/urfave/cli/v2"
+	"os"
+	"os/exec"
+	"time"
+)
+
+// beforeSatelliteAction is run before satelliteAction is run. It is used
+// for checking CLI options.
+func beforeSatelliteAction(ctx *cli.Context) error {
+	// First check if machine-readable format is used
+	err := setupFormatOption(ctx)
+	if err != nil {
+		return err
+	}
+
+	satelliteUrlStr := ctx.String("url")
+	if satelliteUrlStr == "" {
+		return fmt.Errorf("no url provided using --url CLI option")
+	}
+
+	uuid, err := getConsumerUUID()
+	if err != nil {
+		return fmt.Errorf("unable to get consumer UUID: %s", err)
+	}
+	if uuid != "" {
+		return fmt.Errorf("cannot configure connected system to use satellite; disconnect system firtst")
+	}
+
+	return checkForUnknownArgs(ctx)
```

Comment:
I don't know if I've commented on this before, but I like this pattern of returning a call through another function. This function is named very well and this clearly describes what happens through this function. Nicely written.